### PR TITLE
test(zone.js): add zone.js externs test

### PR DESCRIPTION
**Pull request by JiaLiPassion**
Add back the zone.js externs file test for google closure compiler.
The test compiles a test program with and without `zone_externs`.

1. With `zone_externs`, the code should keep the APIs defined in the `zone_externs`.
2. Without `zone_externs`, the code will not keep the APIs

### CODE REVIEW

**Review by vhnguyen28 — APPROVED** [review]

File: packages/zone.js/test/closure/zone.closure.ts [hunk 1]
```diff
@@ -137,6 +172,9 @@ const testClosureFunction = () => {
       result = false;
     }
   }
+  if (result) {
+    console.log('All tests passed');
+  }
```
Comment (mhevery) [inline packages/zone.js/test/closure/zone.closure.ts:177]: maybe also print `console.error` when it does not pass?

**Review by mhevery — APPROVED** [review]

File: packages/zone.js/test/closure/zone.closure.ts (hunk 1 above)
Comment (JiaLiPassion) [reply packages/zone.js/test/closure/zone.closure.ts:179]: Got it, I just updated it.

File: packages/zone.js/package.json [hunk 2]
```diff
@@ -7,19 +7,21 @@
   "es2015": "./fesm2015/zone.js",
   "fesm2015": "./fesm2015/zone.js",
   "typings": "./zone.d.ts",
-  "dependencies": {
-    "tslib": "^2.0.0"
```
Comment (gkalpak) [inline packages/zone.js/package.json:11]: OOC, why this change? Isn't this a breaking change?

File: packages/zone.js/test/closure/zone.closure.ts [hunk 3]
```diff
@@ -72,16 +72,57 @@ const testClosureFunction = () => {
...
+        'runTask',
+        'scheduleTask',
+        'scheduleMicroTask',
+        'scheduleMacroTask',
+        'scheduleEventTask',
+        'cancelTask',
+      ];
+      zonePrototypeKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor((Zone as any).prototype, key);
+        if (desc) {
```
Comment (gkalpak) [inline packages/zone.js/test/closure/zone.closure.ts:91]: Would `hasOwnProperty()` be simpler?

```suggestion
        if ((Zone as any).prototype.hasOwnProperty(key)) {
```

File: packages/zone.js/test/closure/zone.closure.ts [hunk 4]
```diff
@@ -72,16 +72,57 @@ const testClosureFunction = () => {
...
+        'onInvoke',
+        'onHandleError',
+        'onScheduleTask',
+        'onInvokeTask',
+        'onCancelTask',
+        'onHasTask',
+      ];
+      zoneSpecKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor(testZoneSpec, key);
+        if (desc) {
```
Comment (gkalpak) [inline packages/zone.js/test/closure/zone.closure.ts:110]: Would `hasOwnProperty()` be simpler?

```suggestion
        if (testZoneSpec.hasOwnProperty(key)) {
```

File: packages/zone.js/test/closure/zone.closure.ts [hunk 5]
```diff
@@ -72,16 +72,57 @@ const testClosureFunction = () => {
...
+        'onHasTask', 'runCount', 'type', 'source', 'data', 'scheduleFn', 'cancelFn', 'callback',
+        'invoke'
+      ];
+
       const task = Zone.current.scheduleMicroTask('testTask', () => {}, undefined, () => {});
-      Object.keys(task).forEach(key => {
-        logs.push(key);
+      zoneTaskKeys.forEach(key => {
+        const desc = Object.getOwnPropertyDescriptor(task, key);
+        if (desc) {
```
Comment (gkalpak) [inline packages/zone.js/test/closure/zone.closure.ts:123]: Would `hasOwnProperty()` be simpler?

```suggestion
        if (task.hasOwnProperty(key)) {
```

File: packages/zone.js/test/closure/zone.closure.ts [hunk 6]
```diff
@@ -137,6 +172,11 @@ const testClosureFunction = () => {
       result = false;
     }
   }
+  if (result) {
+    console.log('All tests passed.');
+  } else {
+    console.error('Test failed, some public APIs can not be found after closure compiler.');
```
Comment (gkalpak) [inline packages/zone.js/test/closure/zone.closure.ts:178]:
```suggestion
    console.error('Test failed, some public APIs cannot be found after closure compiler.');
```

**Review by gkalpak — APPROVED** [review]
I don't understand 100% of the changes, but since this about tests and CI is happy 🤷‍♂️ :smiley:
Mocing `tslib` to devDependencies is my main concern/question. Otherwise lgtm :+1:

File: packages/zone.js/package.json (hunk 2 above)
Comment (JiaLiPassion) [reply packages/zone.js/package.json:11]: Got it, thank you for pointing out this, I move it back.

File: packages/zone.js/test/closure/zone.closure.ts (hunk 6 above)
Comment (JiaLiPassion) [reply packages/zone.js/test/closure/zone.closure.ts]: Got it, thanks!

File: packages/zone.js/test/closure/zone.closure.ts (hunk 3 above)
Comment (JiaLiPassion) [reply packages/zone.js/test/closure/zone.closure.ts]: Yes, you are right, thanks

File: packages/zone.js/test/closure/zone.closure.ts (hunk 4 above)
Comment (JiaLiPassion) [reply packages/zone.js/test/closure/zone.closure.ts]: Yes, you are right.

File: packages/zone.js/test/closure/zone.closure.ts (hunk 5 above)
Comment (JiaLiPassion) [reply packages/zone.js/test/closure/zone.closure.ts]: Yes, you are right.
